cmd/traffic/cmd/manager/manager.go

```diff
@@ -136,9 +137,13 @@ func (m *Manager) runSystemAGCLoop(ctx context.Context) error {
 					}
 
 					// Release the connection we got to delete the domain + intercept
-					if err := systema.Done(); err != nil {
+					if err := systema.Done(ctx); err != nil {
 						dlog.Errorln(ctx, "systema: release management connection:", err)
 					}
+					// Double free needed to decrease refcount of connection acquired, but not released, in UpdateIntercept
```

Comment:
Where's the call to `UpdateIntercept`? And why isn't `UpdateIntercept` releasing the connection that it acquired?

Reply:
So it's not a call that comes from here, but rather the original UpdateIntercept call that created the domain --it [acquires a connection](https://github.com/telepresenceio/telepresence/blob/release/v2/cmd/traffic/cmd/manager/service.go#L466) but doesn't drop it except [for](https://github.com/telepresenceio/telepresence/blob/release/v2/cmd/traffic/cmd/manager/service.go#L508-L510) [errors](https://github.com/telepresenceio/telepresence/blob/release/v2/cmd/traffic/cmd/manager/service.go#L541-L543). I dug through the history and this seems to be deliberate: the pool's .Get() sets up a reverse connection to systema, so that has to stay open as long as there's an intercept with a preview url live

Reply:
So the `update` that is read from the `snapshot.Updates` "carries" an open connection with it (the comment could be clearer about that)? Isn't there a risk for a leak here? What if the `systema.Get` fails on line 122 for instance? Shouldn't the second `system.Done` be called regardless?

Reply:
> So the update that is read from the snapshot.Updates "carries" an open connection with it (the comment could be clearer about that)?

yes it basically does --for it to get here that connection must've been opened by an updateintercept rpc

> Isn't there a risk for a leak here? What if the systema.Get fails on line 122 for instance? Shouldn't the second system.Done be called regardless?

yes great point, i'll go change this